cmd/juju/application/addunit.go

```diff
@@ -35,6 +36,10 @@ with any application or model constraints. This command also supports the
 placement directive ("--to") for targeting specific machines or containers,
 which will bypass application and model constraints.
 
+Note:
```

Comment:
As with comment in remove-unit PR, I think we need to explain up front that k8s and cloud models are different and preface a block of text for each type.